tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5377,6 +5377,10 @@ class HLSLExternalSource : public ExternalSemaSource {
         return true;
       }
       QualType argType = arg.getAsType();
+      // Allow TemplateTypeParmType for case like
+      // template <typename T> void foo(GroupNodeInputRecords<T> data) {}
+      if (argType->getAs<TemplateTypeParmType>())
```

Comment:
Couldn't we check for `argType->isDependentType()` instead?

Reply:
Yes. isDependentType works.
Hit other issue in test.
Debugging.

Reply:
Was the issue resolved?

Reply:
Yes. It is resolved.